src/category_theory/groupoid/subgroupoid.lean

```diff
@@ -0,0 +1,467 @@
+{ obj := λ s, ⟨s.val, le_carrier h s.prop⟩,
+  map := λ s t f, ⟨f.val, h f.prop⟩,
+  map_id' := λ _, rfl,
+  map_comp' := λ _ _ _ _ _, rfl }
+
+lemma coe_le_inj_on_objects {S T : subgroupoid C} (h : S ≤ T) :
+  function.injective (coe_le h).obj :=
+begin
+  dsimp only [coe_le],
+  rintros ⟨s,hs⟩ ⟨t,ht⟩ he,
+  simp only [subtype.mk_eq_mk] at he ⊢,
+  exact he,
+end
+
+lemma coe_le_faithful {S T : subgroupoid C} (h : S ≤ T) (s t : S.coe):
+  function.injective (λ (f : s ⟶ t), (coe_le h).map f) :=
+begin
+  dsimp only [coe_le],
+  rintros ⟨f,hf⟩ ⟨g,hg⟩ he,
+  simp only [subtype.mk_eq_mk] at he ⊢,
+  exact he,
+end
+
+lemma coe_le_refl {S : subgroupoid C} : coe_le (le_refl S) = 𝟭 S.coe :=
+begin
+  dsimp only [coe_le],
+  fapply functor.ext,
+  { rintros,
+    simp only [subtype.val_eq_coe, subtype.coe_eta, functor.id_obj], },
+  { rintros ⟨s,hs⟩ ⟨t,ht⟩ ⟨f,hf⟩,
+    simp only [eq_to_hom_refl, functor.id_map, category.comp_id, category.id_comp,
+               subtype.mk_eq_mk], }
+end
+
+lemma coe_le_trans {R S T : subgroupoid C} (k : R ≤ S) (h : S ≤ T) :
+  coe_le (le_trans R S T k h) = (coe_le k) ⋙ (coe_le h) :=
+begin
+  dsimp only [coe_le],
+  fapply functor.ext,
+  { rintros, simp only [functor.comp_obj], },
+  { rintros ⟨s,hs⟩ ⟨t,ht⟩ ⟨f,hf⟩,
+    simp only [eq_to_hom_refl, functor.comp_map, category.comp_id, category.id_comp,
+    subtype.mk_eq_mk], }
+end
+
+lemma coe_le_comp_embedding {S T : subgroupoid C} (h : S ≤ T) :
+  (coe_le h) ⋙ T.coe_embedding = S.coe_embedding :=
+begin
+  dsimp only [coe_le, coe_embedding],
+  fapply functor.ext,
+  { rintros, simp only [functor.comp_obj, subtype.val_eq_coe], },
+  { rintros ⟨s,hs⟩ ⟨t,ht⟩ ⟨f,hf⟩,
+    simp only [functor.comp_map, subtype.coe_mk, eq_to_hom_refl, category.comp_id,
+    category.id_comp], }
+end
+
+/-- The family of arrows of the discrete groupoid -/
+inductive discrete.arrws : Π (c d : C), (c ⟶ d) → Prop
+| id (c : C) : discrete.arrws c c (𝟙 c)
+
+/-- The only arrows of the discrete groupoid are the identity arrows-/
+def discrete : subgroupoid C :=
+⟨ discrete.arrws,
+  by { rintros _ _ _ hp, induction hp, simp only [inv_eq_inv, is_iso.inv_id], constructor, },
+  by { rintros _ _ _ _ hp _ hq, induction hp, induction hq, rw category.comp_id, constructor,} ⟩
+
+lemma mem_discrete_iff {c d : C} (f : c ⟶ d):
+  (f ∈ (discrete).arrws c d) ↔ (∃ (h : c = d), f = eq_to_hom h) :=
+begin
+  split,
+  { intro hf, induction hf, simp only [eq_self_iff_true, exists_true_left, eq_to_hom_refl], },
+  { rintro ⟨h,he⟩, subst_vars, constructor, }
+end
+
+/-- A subgroupoid is normal if it is “wide” (meaning that its carrier set is all of `C`)
+    and satisfies the expected stability under conjugacy -/
+structure is_normal : Prop :=
+(wide : ∀ c, (𝟙 c) ∈ (S.arrws c c))
+(conj : ∀ {c d} (p : c ⟶ d) (γ : c ⟶ c) (hs : γ ∈ S.arrws c c),
+              ((inv p) ≫ γ ≫ p) ∈ (S.arrws d d))
+(conj' : ∀ {c d} (p : d ⟶ c) (γ : c ⟶ c) (hs : γ ∈ S.arrws c c),
```

Comment:
Also `conj'` can always can be generated from `conj`, and it is a Prop, not a data field, so there is never a need for the user to provide an alternative value of the field (because all proofs of a Prop are defeq). Given this, I think you should just add a lemma `is_normal.conj'` instead of including it as a field.